### PR TITLE
Add support policy

### PR DESCRIPTION
**Pull request by MaksimZhukov**
In scope of this PR, we described the support policy we adhere to and explained the approach we use to build Python.

### CODE REVIEW

**Review by IvanZosimov — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,13 +4,22 @@ File [versions-manifest.json](./versions-manifest.json) contains the list of ava
...
+- Python >= 3.11. The official macOS `universal2` Python binaries are simply downloaded from [python.org](https://www.python.org/ftp/python/) and archived along with the installation script for further distribution and installation.
+
+**Windows:** The official Python executables are simply downloaded from [python.org](https://www.python.org/ftp/python/) and archived along with the installation script for further distribution and installation.
+
+## Support policy
+We are trying to build and release new versions of Python as soon as they are released. Both stable and unstable versions are considered for building and releasing. Please open an issue in [actions/setup-python](https://github.com/actions/setup-python) if any versions are missed.
```
Comment (brcrista) [inline README.md:20]:
```suggestion
We aim to make new versions of Python available as soon as they are released. Both stable and prerelease versions are considered for building and releasing. Please open an issue in [actions/setup-python](https://github.com/actions/setup-python) if a version you're looking for isn't available.
```

File: README.md [hunk 2]
```diff
@@ -4,13 +4,22 @@ File [versions-manifest.json](./versions-manifest.json) contains the list of ava
...
 
-## Adding new versions
-We are trying to build and release new versions of Python as soon as they are released. Please open an issue in [actions/setup-python](https://github.com/actions/setup-python/issues) if any versions are missing.
+When a new version of operating system is released and available for use as a [GitHub hosted runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), we will build the latest existing patch version of Python for all major versions that have not reached EOL. Please see the [status of Python versions](https://devguide.python.org/versions/) for more information about supported versions. All new versions of Python released after the new OS version is added will also be built for it.
```
Comment (brcrista) [inline README.md:22]:
```suggestion
When a new version of an operating system is released and made available for use with [GitHub hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), we will provide the **latest existing patch versions of Python for all major versions that have not reached [end-of-life](https://devguide.python.org/versions/)**.  Any subsequent Python versions will be made available for the new OS as well.
```

File: README.md [hunk 3]
```diff
@@ -4,13 +4,22 @@ File [versions-manifest.json](./versions-manifest.json) contains the list of ava
...
+More versions are available to install on-the-fly through the [`setup-python`](https://github.com/actions/setup-python) action.
+
+## Building installation packages
+**Ubuntu:** The official Python sources are downloaded from [python.org](https://www.python.org/ftp/python/), built with additional configurations using the make tool and archived along with the installation script for further distribution and installation. We build a Python version against all available [Ubuntu versions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).
+
+**macOS:** 
+- Python < 3.11. The official Python sources are downloaded from [python.org](https://www.python.org/ftp/python/), built with additional configurations using the make tool and archived along with the installation script for further distribution and installation. We build a Python version against the oldest available macOS version for backward compatibility. Available macOS GitHub-hosted runners can be found [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).
```
Comment (brcrista) [inline README.md:14]:
```suggestion
- For **Python < 3.11**, the official Python sources are downloaded from [python.org](https://www.python.org/ftp/python/), built with additional configurations using the `make` tool, and archived along with the installation script for further distribution and installation.  For backward compatibility, we build each Python version against the oldest available macOS version. Available macOS GitHub-hosted runners can be found [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).
```

File: README.md [hunk 4]
```diff
@@ -4,13 +4,22 @@ File [versions-manifest.json](./versions-manifest.json) contains the list of ava
...
+- Python >= 3.11. The official macOS `universal2` Python binaries are simply downloaded from [python.org](https://www.python.org/ftp/python/) and archived along with the installation script for further distribution and installation.
```
Comment (brcrista) [inline README.md:15]:
```suggestion
- For **Python >= 3.11**, the official macOS `universal2` Python binaries are simply downloaded from [python.org](https://www.python.org/ftp/python/) and archived along with the installation script for further distribution and installation.
```

File: README.md [hunk 5]
```diff
@@ -4,13 +4,22 @@ File [versions-manifest.json](./versions-manifest.json) contains the list of ava
...
 > Caution: this is prepared for and only permitted for use by actions `runner-images` and `setup-python` action.
 
-**Status**: Currently under development and in use for beta and preview actions.  This repo is undergoing rapid changes.
-
 Some versions are pre-installed on [runner-images](https://github.com/actions/runner-images) images.
-More versions will (soon!) be available to install on-the-fly through the [`setup-python`](https://github.com/actions/setup-python) action.
+More versions are available to install on-the-fly through the [`setup-python`](https://github.com/actions/setup-python) action.
+
+## Building installation packages
+**Ubuntu:** The official Python sources are downloaded from [python.org](https://www.python.org/ftp/python/), built with additional configurations using the make tool and archived along with the installation script for further distribution and installation. We build a Python version against all available [Ubuntu versions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).
```
Comment (brcrista) [inline README.md:11]:
```suggestion
**Ubuntu:** The official Python sources are downloaded from [python.org](https://www.python.org/ftp/python/), built using the `make` tool, and archived along with the installation script for further distribution and installation. We build each Python version against all available [Ubuntu versions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) at the time the Python version is released.
```

**Review by brcrista — COMMENTED** [review]
Looks good!  Proposed a few rewordings for clarity.

**Pull request merged**